File: queue_stack/daily_temperatures/daily_temperatures.go
func dailyTemperatures(temperatures []int) []int {
    n := len(temperatures)
    stack := make([]int, 0)
    result := make([]int, 0)
    
    for i := 0; i < n; i++ {
        result = append(result, 0)
    }
        
    for i := n - 1; i >= 0; i-- {
        for len(stack) > 0 && temperatures[i] >= temperatures[stack[len(stack) - 1]] {
            stack = stack[:len(stack)-1]
        }         
        
        if len(stack) > 0 {
            result[i] = stack[len(stack) - 1] - i
        }
        
        stack = append(stack, i)
        
    } 
    
    return result
}

File: queue_stack/daily_temperatures/daily_temperatures.py
def get_answer(temperatures) -> [int]:
    n = len(temperatures)
    result = [0] * n
    stack = []

    for i in range(n - 1, -1, -1):
        while stack and temperatures[i] >= temperatures[stack[-1]]:
            stack.pop()
        if stack:
            result[i] = stack[-1] - i
        stack.append(i)
    return result

if __name__ == '__main__':
    temperatures = [73,74,75,71,69,72,76,73] 
    
    print(get_answer(temperatures)) # [1,1,4,2,1,1,0,0]

    